src/meta-srv/src/metasrv.rs

```diff
@@ -109,6 +109,10 @@ pub struct MetasrvOptions {
     /// limit the number of operations in a txn because an infinitely large txn could
     /// potentially block other operations.
     pub max_txn_ops: usize,
+
+    /// The etcd connect options.
+    /// Most of the options are from `etcd_client::ConnectOptions`.
+    pub etcd_connect_options: EtcdConnectOptions,
```

Comment:
This option itself can be `Option`? Or we define a `etcd` subsection for metasrv. Extending this option looks like we leak/couple the metasrv with etcd implementation.

Reply:
It seems reasonable. It also reminds me that there is more than one option for metasrv backend storage(file system, memory, etcd, ...). I think we need the `BackendStorageProvider` and etcd is one of the providers. 

For example:

```toml
[storage]
type = "etcd"
addr = "127.0.0.1"
timeout = "5s"
```

What do you think? @fengjiachun

Reply:
> It seems reasonable. It also reminds me that there is more than one option for metasrv backend storage in the future(file system, memory, etcd, ...). I think we need the `BackendStorageProvider` and etcd is one of the providers.
> 
> For example:
> 
> ```toml
> [storage]
> type = "etcd"
> addr = "127.0.0.1:2379"
> timeout = "5s"
> ```
> 
> What do you think? @fengjiachun

+1